### PR TITLE
dotty phase 1

### PR DESCRIPTION
**Pull request by ohze**
References #28835, #28837

I split #28837 to some PRs.
This is the first PR & only:
+ Add dotty support to sbt.
Sbt settings are only changed if `scalaVersion` is dotty, ex by adding `-Dakka.build.scalaVersion=0.24.0-bin-20200408-4cc224b-NIGHTLY` when running sbt.
+ Source & binary compatible changes

**EDIT: I change this PR to only include changes by running scalafix rules**
_(and some obvious sbt changes)_

### CODE REVIEW

**Review by raboof — APPROVED** [review]

File: akka-actor-tests/src/test/scala/akka/actor/DeathWatchSpec.scala [hunk 1]
```diff
@@ -83,7 +83,7 @@ trait DeathWatchSpec { this: AkkaSpec with ImplicitSender with DefaultTimeout =>
 
   "The Death Watch" must {
     def expectTerminationOf(actorRef: ActorRef) =
-      expectMsgPF(5 seconds, actorRef + ": Stopped or Already terminated when linking") {
+      expectMsgPF(5 seconds, "" + actorRef + ": Stopped or Already terminated when linking") {
```
Comment (renatocaval) [inline akka-actor-tests/src/test/scala/akka/actor/DeathWatchSpec.scala:86]: nitpicking: I find it unfortunate from aesthetics point of view that we need now to prepend a "". Maybe better to use string interpolation.

```suggestion
      expectMsgPF(5 seconds, s"$actorRef: Stopped or Already terminated when linking") {
```

Not a requirement for this PR. We can add it later or let it like that. Is more me sharing a thought.

**Review by octonato — APPROVED** [review]
Thanks @ohze. Much easier to review and move forward. 

(left a small nitpicking comment, not a requirement)

File: akka-actor-tests/src/test/scala/akka/actor/DeathWatchSpec.scala (hunk 1 above)
Comment (giabao) [reply akka-actor-tests/src/test/scala/akka/actor/DeathWatchSpec.scala:86]: I also think that and have tried to implement [Any2StringAdd](https://github.com/ohze/scalafix-rules/blob/master/scalafix/rules/src/main/scala/fix/Any2StringAdd.scala) rule using string interpolation but I found its unnecessary complicated. So I switched to the simple solution by prefixing an empty string.

**Review by patriknw — APPROVED** [review]
LGTM, good call to split up in several PR. All good changes here.

**Comment by patriknw** [comment]
Not squashing because there is value in keeping those commits separately for the history.

**Pull request merged**